weblate/formats/exporters.py

```diff
@@ -141,13 +145,21 @@ def add_unit(self, unit):
             output.markfuzzy(True)
         self.storage.addunit(output)
 
-    def get_response(self, filetemplate='{project}-{language}.{extension}'):
-        filename = filetemplate.format(
+    def asTmpFile(self):
+        tmpFilePath = tempfile.TemporaryFile().name
+        with open(tmpFilePath, 'w') as handle:
```

Comment:
There are several problems with way you use temporary files:

* `tempfile.TemporaryFile` already does open the file, there is no reason to open it again. 
* It can easily happen that the tempfile will be removed (as a result of garbage collection) before it will be read later.
* (slightly related to previous) It can also happen that file files will be kept around for way too long.
* Because of reopening files, this will not work on Windows (you can not open already open file there).

Reply:
1. Was not aware of that, thanks.
2. How to make sure it sticks around?
3. Will make sure to close them with `finally` once I finalize this PR.
4. Will this be sorted out once I change the code with regard to 1.?

Reply:
@nijel Pls answer.